swagger.yaml

```diff
@@ -750,7 +750,349 @@ paths:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/VacationRequest'
+        "default":
+          description: Invalid request was sent to the server.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+
+  "/vacationsRequests/{id}":
+    get:
+      operationId: findVacationRequest
+      summary: Finds vacation request
+      description: Finds persisted vacation request
+      tags:
+        - VacationRequests
+      parameters:
+        - name: id
+          required: true
+          in: path
+          description: VacationRequest id
+          schema:
+            type: string
+            format: uuid
+      responses:
+        '200':
+          description: Found vacation request.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/VacationRequest'
+        "default":
+          description: Invalid request was sent to the server.
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/Error"
+    put:
+      operationId: updateVacationRequest
+      summary: Updates vacation request
+      description: Update vacation request
+      tags:
+        - VacationRequests
+      parameters:
+        - name: id
+          required: true
+          in: path
+          description: VacationRequest id
+          schema:
+            type: string
+            format: uuid
+      requestBody:
+        description: Updated vacation request
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/VacationRequest'
+      responses:
+        '200':
+          description: created new vacation request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/VacationRequest'
+        "default":
+          description: Invalid request was sent to the server.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+    delete:
+      operationId: deleteVacationRequest
+      summary: Delete a vacation request.
+      description: Deletes a vacation request.
+      tags:
+        - VacationRequests
+      parameters:
+        - name: id
+          required: true
+          in: path
+          description: VacationRequest id
+          schema:
+            type: string
+            format: uuid
+      responses:
+        "204":
+          description: Empty response indicating successful removal.
+        "default":
+          description: Invalid request was sent to the server
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+  "/vacationRequests/{id}/statuses":
```

Comment:
Not sure are these vacationRequest status thing so needed. Could these be handled some other way ?

Reply:
In the original design it was decided that the vacation status should be a separate entity, this allows us to record multiple vacation "responses" e.g. approval or decline along with accompanying messages/ notes.

Feel free to have a think if you want to change this and we could have further discussion. At least IMO vacationRequestStatuses are fine for now as the UI is already build around them being a separate entity.

Reply:
Then I guess we will keep them for now